web/client/utils/CoordinatesUtils.js

```diff
@@ -1030,6 +1033,37 @@ export const getPolygonFromCircle = (center, radius, units = "degrees", steps =
     return turfCircle(center, radius, {steps, units});
 };
 
+/**
+ * Returns an array of projections
+ * @return {array} of projection Definitions [{code, extent}]
+ */
+export const getProjections = () => {
+    const projections = (getConfigProp('projectionDefs') || []).concat([{code: "EPSG:3857", extent: [-20026376.39, -20048966.10, 20026376.39, 20048966.10]},
+        {code: "EPSG:4326", extent: [-180, -90, 180, 90]}
+    ]);
+    return projections;
+};
+
+/**
+ * Return a projection from a list of projections
+ * @param code {string} code for the projection EPSG:3857
+ * @return {object} {extent, code} fallsback to default {extent: [-20026376.39, -20048966.10, 20026376.39, 20048966.10]}
+ */
+export const getExtentForProjection = (code = "EPSG:3857") => {
+    return getProjections().find(project => project.code === code) || {extent: [-20026376.39, -20048966.10, 20026376.39, 20048966.10]};
+};
+
+/**
+ * Return a boolean to show if a layer fits within a boundary/extent
+ * @param layer {object} to check if fits with in a projection boundary
+ * @return {boolean} true or false
+ */
+export const checkIfLayerFitsExtentForProjection = (layer = {}) => {
+    const crs = layer.bbox?.crs || "EPSG:3857";
+    const { extent } = getExtentForProjection(crs);
+    const [minx, minY, maxX, maxY] = turfBbox({type: 'FeatureCollection', features: layer.features || []});
+    return ((extent[2] >= maxX) && (extent[3] >= maxY) && (extent[1] >= minx) && (extent[2] > minY));
```

Comment:
there is something wrong here
you are not using extent[0] and when comparing to max they should be inside
please add unit test since they are not validating the correct behaviour
add a unit test for a valid case  and many for invalid with only one condition that is failing
```suggestion
    return ((minx >= crsMinX) && (minY >= crsMinY) && (maxX <= crsMaxX) && (maxY <= crsMaxY));
```
please double check it